private/buf/buflsp/symbol.go

```diff
@@ -362,6 +371,46 @@ func renameChangesForReferenceableSymbol(s *symbol, newName string) (map[protoco
 	return changes, nil
 }
 
+// checkRenameConflicts takes the symbol and desired new name and checks if this conflicts
+// with an existing symbol in the same scope and returns an error if a conflict is found.
+func checkRenameConflicts(target *symbol, newName string) error {
+	parent := target.ir.FullName().Parent()
+	if parent != "" {
+		var existing source.Span
+		newFullName := parent.Append(newName)
+		containsFunc := func(s *symbol) bool {
+			existing = s.span
+			return s.ir.FullName() == newFullName
+		}
+		// We check all files in the workspace for a conflict, since a package can span an arbitrary
+		// number of files.
+		// We first check the current symbol's file.
+		if slices.ContainsFunc(target.file.symbols, containsFunc) {
+			return fmt.Errorf(
+				"Renaming %q to %q would conflict with existing symbol at %q:%d:%d",
```

Comment:
not sure if the initial string should be in quotes, maybe just a string here for the file?:

<img width="677" height="37" alt="Image" src="https://github.com/user-attachments/assets/c737b144-9f15-494f-8ae0-6af7d977f0bb" />

```suggestion
				"Renaming %q to %q would conflict with existing symbol at %s:%d:%d",
```